File: framework/test2/void/revstr/Revstr.scala
import soda.scala.unittest._

import scala.reflect.runtime.universe.typeOf

import soda.scala.leetcode._

import collection.mutable
import util.control.Breaks.{break, breakable}
object Solution {
  def reverseString(s: Array[Char]): Unit = {
    var i = 0
    var j = s.length - 1
    while (i < j) {
      val temp = s(i)
      s(i) = s(j)
      s(j) = temp
      i += 1
      j -= 1
    }
  }
}

class Revstr extends (String => String) {
  override def apply(text: String): String = {
    val work = GenericTestWork.create1u(Solution.reverseString)
    work.compareSerial = true
    work.run(text)
  }
}

object Revstr extends App {
  println(new Revstr()(Utils.fromStdin()))
}

File: framework/test2/void/revstr/Revstr.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Function;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.work.TestWork;
import soda.unittest.work.Utils;


class Solution {
    public void reverseString(char[] s) {
        int i = 0, j = s.length-1;
        while (i < j) {
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            ++i;
            --j;
        }
    }
}

public class Revstr implements Function<String, String> {

    @Override
    public String apply(String text) {
        var work = TestWork.forObject(new Solution(), "reverseString");
        // work.setValidator((e, r) -> {...});
        work.setCompareSerial(true);
        return work.run(text);
    }

    public static void main(String[] args) throws Exception {
        System.out.println(new Revstr().apply(Utils.fromStdin()));
    }

}
